sdk/keyvault/azkeys/crypto/crypto_client.go

```diff
@@ -57,17 +57,21 @@ func parseKeyIDAndVersion(id string) (string, string, error) {
 		return "", "", err
 	}
 
-	path := strings.Split(parsed.Path, "/")
+	if !strings.HasPrefix(parsed.Path, "/keys/") {
+		return "", "", fmt.Errorf("could not parse Key ID from %s", id)
```

Comment:
I would change the error message here so it's different from line 67 (right now the message is ambiguous).